src/main/java/io/citrine/jcc/search/analysis/result/AnalysisResultWithBuckets.java

```diff
@@ -0,0 +1,164 @@
+package io.citrine.jcc.search.analysis.result;
+
+import io.citrine.jcc.util.ListUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Base class for all results of analysis operations that produce buckets.
+ *
+ * @param <T> Type of the bucket.
+ * @author Kyle Michel
+ */
+public abstract class AnalysisResultWithBuckets<T extends AnalysisResultWithBuckets.Bucket> extends AnalysisResult {
+
+    /**
+     * Set the list of buckets. This replaces any buckets that are already present.
+     *
+     * @param buckets List of bucket objects.
+     * @return This object.
+     */
+    public AnalysisResultWithBuckets<T> setBuckets(final List<T> buckets) {
+        this.buckets = buckets;
+        return this;
+    }
+
+    /**
+     * Add to the list of buckets.
+     *
+     * @param buckets List of bucket objects.
+     * @return This object.
+     */
+    public AnalysisResultWithBuckets<T> addBuckets(final List<T> buckets) {
+        this.buckets = ListUtil.add(buckets, this.buckets);
+        return this;
+    }
+
+    /**
+     * Add to the list of buckets.
+     *
+     * @param buckets Bucket object to add.
+     * @return This object.
+     */
+    public AnalysisResultWithBuckets<T> addBuckets(final T buckets) {
```

Comment:
`addBucket`?

Reply:
The convention that I've used in the past in this package (which you may disagree with) is to use {set, add, get}FieldName for setters, adders, and getters, ignoring the arguments to the method.